### ISSUE TITLE
git tag patch versions

### ISSUE THREAD

**Issue by firmart** [tdlib/td#1627]
Sometimes I want to compile specific TDlib version as precise as patch version (e.g.  1.7.4). Would it be possible to `git tag` those versions in the future ? It can also be done easily retroactively: `git log --grep="version" -- :CMakeLists.txt`.

**Comment by levlam** [comment]
It would be wrong to tag these commits. Telegram is developed continuously and those aren't the commits you are looking for.

The TDLib version is increased the same day a Telegram update is released to denote significant changes in the TDLib API. This happens along with push of source code with all new features inplemented. After that there can be a lot of improvements and bug fixes before the next release. Therefore the exact commit, in which the version was changed, isn't recommended for usage to anyone. In fact, if you want to use the most stable TDLib 1.X.Y, you need to search for the first commit with that version and message "Update/Increase layer to ..." and use the commit just before that.

But everyone should tend to use as new TDLib version is possible. Telegram updates frequently and new features aren't available in old TDLib versions. Moreover, due to server-side changes or introduction of new features some previously working features can be broken.

So, basically, each time a version is stable enough and can be marked with a tag, there is a newer version with the latest features, which should be used instead. Most of the time, the current master version is the TDLib version, which should be used.

**Comment by firmart** [comment]
Ok, it's fair enough.

**Comment by asarubbo** [comment]
Hello @levlam 

I never asked to tag each commit , but I asked to release frequently, that is a bit different.

You should see the situation from a packager perspective where sees a tag as a point of reference, while the master version is always seen as 'unstable' version.

See also: https://github.com/boto/botocore/tags

Anyway I'm using [python-telegram](https://github.com/alexander-akhmetov/python-telegram) that uses a bundled version of [tdlib-1.8.5](https://github.com/alexander-akhmetov/python-telegram/tree/839835228f9c4311182a28b0b6b2b78527647fac/telegram/lib/linux) and it works for me.

By digging a bit into the History of CMakeLists.txt I can see that 1.8.5 was 'added' on [Jul 31, 2022 ](https://github.com/tdlib/td/commit/d9cfcf88fe4ad06dae1716ce8f66bbeb7f9491d9) so the facts says that a version from 10 months ago is still working.
Based on that I didn't really understand your idea behind not make tags, but more than this I don't understand the version increased in CMakeLists.txt **without** a tag.

**Comment by levlam** [comment]
For TDLib master version is always suitable for production releases. And this is always the recommended for use version, because all app users are supposed to update frequently and use the latest version. The old versions will still work, but they don't have support for all features and may be otherwise limited.

**Comment by asarubbo** [comment]
I understand your concern and you are certainly right.

What I am asking (but not only me, I see all the discussions about the tags) is at least tag when you 'declare' a new version like you did here:
https://github.com/tdlib/td/commit/8517026415e75a8eec567774072cbbbbb52376c1
https://github.com/tdlib/td/commit/eb664b4e9f71945ef3faa802af779e1630fedb8f
https://github.com/tdlib/td/commit/0c09070ce5d36bcccc5533c89dea45909483f45a

So at least packagers can properly use [repology](https://repology.org/project/tdlib-telegram/versions) instead of manually monitor the git commits.

Thanks

**Comment by njfamirm** [comment]
I'm create tag and release in this repository: https://github.com/njfamirm/td

**Comment by 667bdrm** [comment]
> So, basically, each time a version is stable enough and can be marked with a tag, there is a newer version with the latest features, which should be used instead. Most of the time, the current master version is the TDLib version, which should be used.

But I know the situations when the client used 'latest' stable tag 1.8.0 with tdlib-example based client and session kicked with UPDATE_APP_TO_LOGIN (and session lost permanent since porting and recompiling it with latest tdlib master branch does not help.

**Comment by levlam** [comment]
@667bdrm 
The latest stable version is always the latest version on Github.
A session can't be kicked with the error "UPDATE_APP_TO_LOGIN", which can be received only during logging in. The client being logged out doesn't depend on the app version, only on the behavior of the app. The error can't be received in the latest TDLib version from master branch.
